src/microsoft.sheets/spreadsheet_endpoint.bal

```diff
@@ -0,0 +1,579 @@
+            "was passed as the table name (" + tableName + "). Now patching the table with the correct " +
+            "table name.");
+
+        Error? renameResult = resultsTable->renameTable(tableName);
+
+        if (renameResult is ()) {
+            return resultsTable;
+        } else {
+            return TableError("Error ocurred while renaming the created table.", renameResult);
+        }
+    }
+
+    # Open a Table.
+    # + tableName - name of the table to be opened
+    # + return - A Table client object on success, else returns an error
+    public remote function openTable(string tableName) returns @tainted (Table|Error) {
+        //Make a GET request and retrieve the table information
+        http:Request request = new;
+        http:Response|error response = self.worksheetClient->get(<@untainted>("/v1.0/me/drive/root:" + self.properties.path +
+            self.properties.workbookName + ".xlsx:/workbook/worksheets/" +
+            self.properties.worksheetName + "/tables/" + tableName), request);
+
+        if (response is error) {
+            return HttpError("Error occurred while accessing the Microsoft Graph API.", response);
+        }
+
+        http:Response httpResponse = <http:Response>response;
+
+        if (httpResponse.statusCode != http:STATUS_OK) {
+            return HttpResponseHandlingError("Error occurred while inserting data into table.");
+        }
+
+        //If the table exists we will get a JSON response with the table's information
+        json|error responseJson = httpResponse.getJsonPayload();
+
+        if !(responseJson is map<json>) {
+            typedesc<any|error> typeOfRespone = typeof responseJson;
+            return TypeConversionError("Invalid response; expected a `map<json>` found " + typeOfRespone.toString());
+        }
+
+        map<json> payload = <map<json>>responseJson;
+
+        json|error identifier = payload.id;
+        if !(identifier is string) {
+            typedesc<any|error> typeOfIdentifier = typeof identifier;
+            return TypeConversionError("Invalid identifier; expected a `string` found " + typeOfIdentifier.toString());
+        }
+
+        string sheetIdentifier = <string>identifier;
+
+        //Address is not returned from the above API call. Hence the address is initialized to an empty string
+        string address = "";
+
+        Table resultsTable = <@untainted>new (self.microsoftGraphConfig, self.properties.path,
+            self.properties.workbookName, self.properties.sheetId, self.properties.worksheetName,
+            sheetIdentifier, address, tableName);
+
+        return resultsTable;
+    }
+};
+
+# Table Client Object.
+# + tableClient - HTTP client endpoint for the table
+# + properties - table specific properties
+public type Table client object {
+    http:Client tableClient;
+    TableProperties properties;
+
+    public function init(MicrosoftGraphConfiguration msGraphConfig, string path, string workbookName, string sheetId,
```

Comment:
Same as before, can't we pass the `httpClient` we created before

Reply:
We can. Refactored the code to use the same HTTP client across all the client objects.